src/du_parameters/initial_bwp.py

```diff
@@ -0,0 +1,123 @@
+    """Exception raised when calculation of BWP location and bandwidth fails."""
+
+    pass
+
+
+def get_total_prbs_from_scs(subcarrier_spacing: Frequency) -> int:
+    """Get the number of Physical Resource Blocks (PRBs) based on Subcarrier Spacing (SCS).
+
+    Args:
+        subcarrier_spacing (Frequency): Subcarrier Spacing as Frequency object.
+        Valid values: 15, 30, 60, 120.
+
+    Returns:
+        int: Number of PRBs for given Subcarrier spacing value.
+
+    Raises:
+        ValueError: If the Subcarrier spacing value is not supported.
+        TypeError: If the Subcarrier spacing is not a Frequency object.
+    """
+    if not isinstance(subcarrier_spacing, Frequency):
+        raise TypeError(
+            f"Subcarrier spacing must be a Frequency object, not {type(subcarrier_spacing)}"
+        )
+
+    scs_to_prbs = {
+        Frequency.from_khz(15): 275,
+        Frequency.from_khz(30): 137,
+        Frequency.from_khz(60): 69,
+        Frequency.from_khz(120): 33,
+    }
+    try:
+        return scs_to_prbs[subcarrier_spacing]
+    except KeyError:
+        supported_subcarrier_spacing = ", ".join(str(freq) for freq in scs_to_prbs.keys())
+        logger.error(
+            "Subcarrier spacing value: %s is not supported. Supported values: %s",
+            subcarrier_spacing,
+            supported_subcarrier_spacing,
+        )
+        raise ValueError(
+            f"Subcarrier spacing value {subcarrier_spacing} is not supported."
+            f"Supported values: {supported_subcarrier_spacing}"
+        )
+
+
+def calculate_initial_bwp(carrier_bandwidth: int, subcarrier_spacing: Frequency) -> int:
+    """Calculate the initial BWP location and bandwidth.
+
+    Uses carrier bandwidth and Subcarrier Spacing.
+
+    Formula:
+        initialBWPlocationAndBandwidth (int) = (Total_PRB * (L - 1)) + RBstart
+        RBstart (int) = 0 the starting resource block for initial BWP location
+        Total_PRB (int) = Total number of Physical Resource Blocks (PRBs) based
+         on Subcarrier Spacing (SCS)
+        L (int) = Carrier bandwidth
+
+    Args:
+        carrier_bandwidth (int): Channel bandwidths of the carrier in terms of Resource Blocks.
+                                Indicated as L in the formula.
+        subcarrier_spacing (Frequency): Subcarrier Spacing in Frequency object.
+
+    Returns:
+        int: The calculated initialBWP location and bandwidth.
+    """
+    if not isinstance(subcarrier_spacing, Frequency):
+        logger.error(
+            "Subcarrier spacing must be a Frequency object, not %s", type(subcarrier_spacing)
+        )
+        raise TypeError(
+            f"Subcarrier spacing must be a Frequency object, not {type(subcarrier_spacing)}"
+        )
+
+    if not isinstance(carrier_bandwidth, int):
+        logger.error("Carrier bandwidth must be an integer, not %s", type(carrier_bandwidth))
+        raise TypeError(f"Carrier bandwidth must be an integer, not {type(carrier_bandwidth)}")
+
```

Comment:
```suggestion
    if not isinstance(subcarrier_spacing, Frequency):
        logger.error(
            "Subcarrier spacing must be a Frequency object, not %s", type(subcarrier_spacing)
        )
        raise TypeError(
            f"Subcarrier spacing must be a Frequency object, not {type(subcarrier_spacing)}"
        )

    if not isinstance(carrier_bandwidth, int):
        logger.error("Carrier bandwidth must be an integer, not %s", type(carrier_bandwidth))
        raise TypeError(f"Carrier bandwidth must be an integer, not {type(carrier_bandwidth)}")

```
I'd remove those, we are using type hints already.

Reply:
Yes, there are type hints but they do not prevent us to call this function with invalid argument types.  An invalid subcarrier spacing raise error but  carrier_bandwidth may not raise. For example if it is  float value, function will return float which is not desired.

Reply:
@Gmerold , @patriciareinoso what do you think ?